lib/modules/llm_service.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+class LlmService
+  def initialize
+    puts 'please provide child class'
+  end
+
+  def llm_object_attributes
+    {
+      scrape_or_process: @scrape_or_process,
+      model: @params[:model],
+      prompt: @prompt,
+      input: @input,
+      output: @output,
+      needs_processing: false
+    }
+  end
+
+  def unload_json(event, response)
+    response_json = JSON.parse(response)
+    response_json.each_key do |key|
+      event[key] = response_json[key]
+    end
+    event
+  end
+
+  def scrape(event_page)
+    @scrape_or_process = 'scrape'
+    @prompt = File.read('lib/modules/llm_prompts/llm_scrape_prompt.txt')
+    @input = event_page
+    content = @prompt.gsub('*replace_with_event_page*', event_page)
+    @output = run(content)
+    @output
+  end
+
+  def process(event)
+    @scrape_or_process = 'process'
+    event_json = JSON.generate(event.to_json)
+    @prompt = File.read('lib/modules/llm_prompts/llm_process_prompt.txt')
+    @input = event_json
+    content = @prompt.gsub('*replace_with_event*', event_json)
+    @output = run(content)
+    @output
+  end
+
+  def scrape_func(event, event_page)
+    response = scrape(event_page)
+    event = unload_json(event, response)
+    event.llm_object_attributes = llm_object_attributes
+    event
+  end
+
+  def post_process_func(event)
+    response = process(event)
+    event = unload_json(event, response)
+    event.llm_object_attributes = llm_object_attributes
```

Comment:
This overwrites the data from the "scrape" LLM interaction, right? Is there ever a need to retain the data for both?

Reply:
Currently not I think. I'd like to still have data for the scrape in case we ever decide not to do post processing for whatever reason